src/ansys/fluent/core/quantity.py

```diff
@@ -0,0 +1,184 @@
+
+import pint
+from pint import Unit
+
+base_units = pint.UnitRegistry(autoconvert_offset_to_baseunit=True)
+base_units.default_system = "SI"
+
+# Unit definitions
+base_units.define("1.e15-particles/kg = 1 kg^-1 = 1^15particles kg^-1")
+base_units.define("1.e15-particles m^-3 s^-1 = 1 m^-3 s^-1 = 1^15particles m^-3 s^-1")
+base_units.define("ton_force = 1 UK_force_ton = tonf")
+
+
+quantity = base_units.Quantity
+
+
+restricted_units = ["Hz", "hertz", "rad/s", "radian/s", "rpm", "rps", "cps"]
+restricted_conversions = list((permutations(restricted_units, 2)))
+
+
+std_prefixes = [
+    "y",
+    "z",
+    "a",
+    "f",
+    "p",
+    "n",
+    "u",
+    "c",
+    "d",
+    "da",
+    "h",
+    "m",
+    "k",
+    "M",
+    "G",
+    "T",
+    "P",
+    "E",
+    "Z",
+    "Y",
+    "",
+]
+
+
+restricted_unit_expansion = {
+    "Hz": [prefix + "Hz" for prefix in std_prefixes],
+    "hertz": [prefix + "hertz" for prefix in std_prefixes],
+    "rad/s": ["radian/s", "rad/s"],
+    "radian/s": ["rad/s", "radian/s"],
+    "rpm": ["revolutions_per_minute", "rpm"],
+    "rps": ["revolutions_per_second", "rps"],
+    "cps": ["counts_per_second", "cps"],
+}
+
+
+def build_restricted_conversions(conversions, unit_expansion):
+
+    """This function generates required final restricted mappings."""
+
+    keys = set([i[0] for i in conversions])
+    restricted_units_dict = {unit[0]: [] for unit in conversions}
+
+    for key in keys:
+        temp_list = [unit_expansion[i] for i in keys.difference(set([key]))]
+        for temp in temp_list:
+            restricted_units_dict[key] += temp
+    for key in keys:
+        restricted_units_dict[key] = list(set(restricted_units_dict[key]))
+    return restricted_units_dict
+
+
+restricted_units = build_restricted_conversions(
+    restricted_conversions, restricted_unit_expansion
+)
+
+
+class Quantity:
+
+    """This class instantiates physical quantities using their real values and
+    units.
+
+    All the instances of this class are converted to base SI units
+    system. Any conversion between "Hz", "hertz", "rad/s", "radian/s",
+    "rpm", "rps", "cps" is restricted.
+    """
+
+    def __init__(self, real_value, units_string):
+        self.value = real_value
+        self.unit = units_string
+        self._quantity = quantity(self.value, self.unit)
+        self._base_si_quantity = self._quantity.to_base_units()
+        self._restricted_conversions = restricted_units
+
+    def __float__(self):
+        return Quantity(self.value, self.unit)
+
+    def __str__(self):
+        return f'({self.value}, "{self.unit}")'
+
+    def __repr__(self):
+        return f'(Quantity ({self.value}, "{self.unit}"))'
+
+    def to(self, to_unit):
+
+        """This method checks the compatibility between current instance unit
+        and user provided unit, if both of them are compatible, then only it
+        performs required conversion otherwise raises a Value Error."""
+
+        if (
```

Comment:
@hpohekar in fact, what we are going to need to do here is to get the dimensions of each units. Each dimensions can be represented as a numerical sequence.  I hope that pint will give us a method to get the dimensions from the units and that that dimensions object includes a field for angle. If so, we only need to compare those two dimensions objects. I'm sure that pint can provide that. That will be rigorous. The current approach cannot unfortunately extend to cover every possibility.

Reply:
Yes pint supports method for dimensionality.

![image](https://user-images.githubusercontent.com/106588300/186197828-bec65c9d-0012-4717-ba13-1eb1f1d3ed22.png)

Reply:
@hpohekar can you please check also the dimensions for radians. I'm sure it's going to be correct. It means all that complicated, incomplete code can go away.

Reply:
![image](https://user-images.githubusercontent.com/106588300/186349372-f817bf80-14af-40a0-a04a-5fef5cfa8105.png)

Reply:
There is no reserved field for angle in dimensions and hence we can not compare 2 dimensions.